src/api_server/src/request/snapshot.rs

```diff
@@ -5,9 +5,23 @@ use super::super::VmmAction;
 use crate::parsed_request::{Error, ParsedRequest};
 use crate::request::Body;
 use crate::request::{Method, StatusCode};
-use vmm::vmm_config::snapshot::{CreateSnapshotParams, LoadSnapshotParams};
+use logger::{IncMetric, METRICS};
+use serde::de::Error as DeserializeError;
+use vmm::vmm_config::snapshot::{
+    CreateSnapshotParams, LoadSnapshotConfig, LoadSnapshotParams, MemBackendConfig, MemBackendType,
+};
 use vmm::vmm_config::snapshot::{Vm, VmState};
 
+/// Deprecation message for the `mem_file_path` field.
+const LOAD_DEPRECATION_MESSAGE: &str = "PUT /snapshot/load: mem_file_path field is deprecated.";
+/// None of the `mem_backend` or `mem_file_path` fields has been specified.
+pub const MISSING_FIELD: &str =
+    "missing field: either `mem_backend` or `mem_file_path` is required";
+/// Both the `mem_backend` and `mem_file_path` fields have been specified.
+/// Only specifying one of them is allowed.
+pub const TOO_MANY_FIELDS: &str =
+    "too many fields: either `mem_backend` or `mem_file_path` is required";
```

Comment:
nit: Seems a bit unclear right now for someone reading, might also add "exclusively" here to suggest the exclusive or we're trying to impose.

```suggestion
    "too many fields: either `mem_backend` or `mem_file_path` exclusively is required";
```